.github/workflows/pr.yml

```diff
@@ -0,0 +1,73 @@
+name: Validate pull request
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, edited]
+
+jobs:
+  check-pr:
+    name: Validate commit subject lines
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run validation script
+        uses: actions/github-script@v7.0.1
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const pullRequest = {
+              pull_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            };
+
+            const { data: commits } = await github.rest.pulls.listCommits(pullRequest);
+            const { data: reviews } = await github.rest.pulls.listReviews(pullRequest);
+
+            // Get latest review by `github-actions[bot]`
+            const githubActionsBot = 41898282;
+            const { state } = reviews
+              .filter(r => r.user.id === githubActionsBot)
+              .at(-1);
+
+            // Ensure commit subject lines are in the form `area or areas: Description`
+            const subjectLineFormat = /^.+:\s.+$/;
+            const badCommits = commits
+              .map(c => [c.sha.slice(0, 8), c.commit.message.split('\n')[0]])
+              .filter(([_, s]) => !subjectLineFormat.test(s))
+              .map(([sha, title]) => `- ${sha}: ${title}`);
+
+            if (0 < badCommits.length) {
```

Comment:
nit: invert yoda condition

Reply:
it improves the readability of the code, since most of the western world isn't RtL oriented